cmd/goal/interact.go

```diff
@@ -694,19 +693,18 @@ var appQueryCmd = &cobra.Command{
 			}
 
 			// Get creator information
-			ad, err := client.AccountData(app.Params.Creator)
+			resp, err := client.AccountApplicationInformation(app.Params.Creator, appIdx)
```

Comment:
The AccountApplicationInformation and AccountAssetInformation endpoints support msgpack too (returns a ledgercore.AccountResource object), so adding a method to hit the "raw" format like `libgoal.Client:AccountData` does would result in fewer changes to implement type conversions between the JSON generated types and the basics types. Doing this would make us feel a little safer that there were no unexpected differences marshaling and unmarshaling through JSON on its way to goal.

Reply:
I just realized that the raw format still doesn't work since the `AccountResource` type holds a bunch of pointers, and simply encoding with messagepack only encodes the pointers. Any suggestions on how to make the raw AccountApplicationInformation api work properly?